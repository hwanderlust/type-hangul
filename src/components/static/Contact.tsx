import React from "react";
import styled from "styled-components";

import { Disclosure, Logo, MenuBtn } from "../common";

const Title = styled.h1`
  font-family: 'Playfair Display', serif;
  font-size: calc(36px + (72 - 36) * ((100vw - 300px) / (1440 - 300)));
  font-weight: 400;
`;

function Contact() {
  return (
    <>

      <Title>Contact</Title>

      <Logo />
      <MenuBtn />

      <h2>Email</h2>
      <p>Any questions or concerns can be directed to <span>hwanderlustdev@gmail.com</span>.</p>

      <h2>GitHub</h2>
      <p>Any requests or issues can be directed to <span>github.com/hwanderlust</span>.</p>

      <Disclosure />
    </>
  );
}

export default Contact;