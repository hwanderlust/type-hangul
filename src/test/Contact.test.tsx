import 'jest-styled-components';
import React from 'react';
import { BrowserRouter } from "react-router-dom";
import renderer from 'react-test-renderer';

import Contact from "../components/static/Contact";

describe("<Contact/>", () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Contact />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
