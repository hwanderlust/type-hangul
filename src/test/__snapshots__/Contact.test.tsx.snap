// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Contact/> renders correctly 1`] = `
.c14 {
  margin-top: 10vh;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-animation: hUniol 4000ms ease-in;
  animation: hUniol 4000ms ease-in;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c16 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  width: 30vw;
  height: 1px;
  background-color: black;
}

.c17 {
  font-family: 'Roboto',serif;
  font-size: calc(8px + (12 - 8) * ((100vw - 300px) / (1440 - 300)));
  font-style: italic;
  padding: 2vw;
}

.c2 {
  display: block;
}

.c3 {
  height: 100px;
  -webkit-transform: translateY(-10%);
  -ms-transform: translateY(-10%);
  transform: translateY(-10%);
}

.c4 {
  width: 100px;
  height: 100px;
  background: transparent;
  border: none;
  border-radius: 50%;
  cursor: pointer;
  -webkit-filter: drop-shadow(1px 0 10px #000);
  filter: drop-shadow(1px 0 10px #000);
}

.c4:focus {
  border: 1px solid white;
  outline: none;
}

.c5 {
  width: 100%;
  height: 100%;
}

.c0 {
  display: grid;
  grid-template-columns: 48vw 48vw;
  grid-template-rows: 100px auto auto auto;
  grid-column-gap: 2vw;
}

.c1 {
  display: none;
}

.c6 {
  grid-area: 1 / 1 / 3 / 3;
  margin-top: 2vh;
  margin-bottom: 0;
  padding-left: 5vw;
  font-family: 'Playfair Display',serif;
  font-size: calc(36px + (72 - 36) * ((100vw - 300px) / (1440 - 300)));
  font-weight: 400;
}

.c7 {
  grid-area: 3 / 1 / 4 / 2;
  width: 100%;
  margin: auto 0;
  border-top-right-radius: 100px;
  border-bottom-right-radius: 100px;
}

.c8 {
  grid-area: 3 / 2 / 4 / 3;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c9 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: black;
}

.c9:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  -webkit-text-decoration-color: black;
  text-decoration-color: black;
  outline: none;
}

.c10 {
  font-family: 'Playfair Display',serif;
  font-size: calc(24px + (36 - 24) * ((100vw - 300px) / (1440 - 300)));
  margin-bottom: 0;
  font-weight: 400;
}

.c11 {
  font-family: 'Roboto',serif;
  font-size: calc(12px + (16 - 12) * ((100vw - 300px) / (1440 - 300)));
  color: white;
  width: 75%;
}

.c12 {
  color: black;
}

.c13 {
  display: block;
  position: absolute;
  right: 1vw;
  top: calc(50% - 50px);
}

.c15 {
  grid-area: 4 / 1 / 5 / 3;
}

@media screen only and (max-width:720px) {
  .c17 {
    max-width: 500px;
  }
}

@media only screen and (max-width:720px) {
  .c3 {
    height: 50px;
    -webkit-transform: translateY(-25%);
    -ms-transform: translateY(-25%);
    transform: translateY(-25%);
  }
}

@media only screen and (max-width:720px) {
  .c4 {
    width: 50px;
    height: 50px;
  }
}

@media only screen and (max-width:720px) {
  .c0 {
    grid-template-columns: 100%;
    grid-template-rows: 75px auto auto auto;
  }
}

@media only screen and (max-width:720px) {
  .c1 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    padding: 0 2.5vw;
    width: 95vw;
  }
}

@media only screen and (max-width:720px) {
  .c6 {
    grid-area: auto;
    margin-bottom: 32px;
  }
}

@media only screen and (max-width:720px) {
  .c7 {
    grid-area: 2 / 1 / 3 / 2;
    border-top-right-radius: 0;
    border-bottom-right-radius: 0;
  }
}

@media only screen and (max-width:720px) {
  .c8 {
    grid-area: auto;
    padding-left: 5vw;
  }
}

@media only screen and (max-width:720px) {
  .c10 {
    margin-top: 0;
  }
}

@media only screen and (max-width:720px) {
  .c11 {
    margin-top: 0.25rem;
  }
}

@media only screen and (max-width:720px) {
  .c13 {
    display: none;
  }
}

@media only screen and (max-width:720px) {
  .c15 {
    grid-area: 6 / 1 / 7 / 2;
  }
}

<div
  className="c0"
>
  <nav
    className="c1"
  >
    <a
      className="c2"
      href="/"
      onClick={[Function]}
    >
      <img
        alt="a 'T' with the Korean ã…Ž as a logo"
        className="c3"
        src="logo.png"
      />
    </a>
    <div>
      <button
        className="c4"
        onClick={[Function]}
      >
        <img
          alt="Kakao Friends' Apeach smiling - as a menu icon/button"
          className="c5"
          src="apeach-btn.png"
        />
      </button>
    </div>
  </nav>
  <h1
    className="c6"
  >
    Contact
  </h1>
  <img
    alt="Kakao friends' posing for a picture with a temple in the background and the skyblue sky"
    className="c7"
    src="friends-posing.jpg"
  />
  <main
    className="c8"
  >
    <section>
      <a
        className="c9"
        href="mailto:hwanderlustdev@gmail.com?subject=Type%20Hangul%20Inquiry"
      >
        <h2
          className="c10"
        >
          Email
        </h2>
        <p
          className="c11"
        >
          Any questions or concerns can be directed to 
          <span
            className="c12"
          >
            hwanderlustdev@gmail.com
          </span>
          .
        </p>
      </a>
    </section>
    <section>
      <a
        className="c9"
        href="https://github.com/hwanderlust"
      >
        <h2
          className="c10"
        >
          GitHub
        </h2>
        <p
          className="c11"
        >
          Any requests or issues can be directed to 
          <span
            className="c12"
          >
            github.com/hwanderlust
          </span>
          .
        </p>
      </a>
    </section>
  </main>
  <div
    className="c13"
  >
    <button
      className="c4"
      onClick={[Function]}
    >
      <img
        alt="Kakao Friends' Apeach smiling - as a menu icon/button"
        className="c5"
        src="apeach-btn.png"
      />
    </button>
  </div>
  <footer
    className="c14 c15"
  >
    <div
      className="c16"
    />
    <p
      className="c17"
    >
      Disclosure: The use of Kakao characters and images are not intended for commercial use and only for educational purposes. No intentions of copyright infingement whatsoever.
    </p>
  </footer>
</div>
`;
